redislock.go

```diff
@@ -49,13 +50,22 @@ func New(client RedisClient) *Client {
 // Obtain tries to obtain a new lock using a key with the given TTL.
 // May return ErrNotObtained if not successful.
 func (c *Client) Obtain(ctx context.Context, key string, ttl time.Duration, opt *Options) (*Lock, error) {
-	// Create a random token
-	token, err := c.randomToken()
-	if err != nil {
-		return nil, err
+	var (
+		value, token string
+		err          error
+	)
+
+	if opt.getSecret() != "" {
+		token = opt.getSecret()
```

Comment:
Sorry, but how does is this supposed to work? This would allow you to obtain a lock 2x, which is exactly what you would want to avoid?